lute/static/js/dict-tabs.js

```diff
@@ -0,0 +1,445 @@
+    activateTab(tab0, dictTabButtons);
+  });
+}
+
+function getSentenceURL() {
+  const txt = TERM_FORM_CONTAINER.querySelector("#text").value;
+  // check for the "new term" page
+  if (txt.length == 0) return;
+  // %E2%80%8B is the zero-width string.  The term is reparsed
+  // on the server, so this doesn't need to be sent.
+  const t = encodeURIComponent(txt).replaceAll('%E2%80%8B', '');
+  if (LANG_ID == '0' || t == '')
+    return;
+
+  const url = `/term/sentences/${LANG_ID}/${t}`;
+
+  return url;
+}
+
+function activateTab(tab, allTabs) {
+  const iFrame = allTabs.get(tab);
+  allTabs.forEach((iframe, btn) => {
+    btn.classList.remove("dict-btn-active");
+    if (iframe) iframe.classList.remove("dict-active");
+  });
+  
+  tab.classList.add("dict-btn-active");
+  if (iFrame) iFrame.classList.add("dict-active");
+}
+
+function loadDictPage(dictID, iFrame) {
+  //const iFrameName = iFrame.getAttribute("name") ? iFrame : "";
+  const term = TERM_FORM_CONTAINER.querySelector("#text").value;
+
+  if (dictID == -1) {
+    do_image_lookup(term, iFrame);
+  } else {
+    const dict = TERM_DICTS[dictID];
+    //const is_external = (activeTab.dataset.dictExternal == 1) ? true : false;
+    show_lookup_page(dict, term, iFrame);
+  }
+}
+
+function createIFrame(name, parent) {
+  const iFrame = document.createElement("iframe");
+  iFrame.name = name;
+  iFrame.src = "about:blank";
+  iFrame.classList.add("dictframe");
+
+  parent.appendChild(iFrame);
+
+  return iFrame;
+}
+
+function createTabBtn(label, parent, data, external, faviconURL=null) {
+  const btn = document.createElement("button");
+  if (label) {
+    btn.textContent = label;
+    btn.setAttribute("title", label);
+  }
+  if (data != null) btn.dataset.dictId = data;
+  if (external != null) {
+    btn.dataset.dictExternal = external;
+    if (external == 1) {
+      const arrowEl = createImg("", "dict-btn-external-img");
+      btn.appendChild(arrowEl);
+    }
+  }
+
+  btn.classList.add("dict-btn");
+
+  if (faviconURL) {
+    const faviconEl = createImg(faviconURL, "dict-btn-fav-img");
+    btn.prepend(faviconEl);
+  }
+  
+  parent.appendChild(btn);
+
+  return btn;
+}
+
+function createImg(src, className) {
+  const img = document.createElement("img");
+  img.classList.add(className);
+  if (src) img.src = src;
+
+  return img;
+}
+
+function isURLExternal(dictURL) {
+  return (dictURL.charAt(0) == '*') ? 1 : 0; 
+}
+
+function getDictURLDomain(url) {
+  const cleanURLString = url.split("*").splice(-1)[0];
+  const urlObj = new URL(cleanURLString);
+
+  return urlObj.hostname;
+}
+
+function getFavicon(domain) {
+  return `http://www.google.com/s2/favicons?domain=${domain}`;
+}
+
+/**
+   * Either open a new window, or show the result in the correct frame.
+   */
+function show_lookup_page(dicturl, text, iframe) {
+  // if iframe is provided use that, else it's an external link
+
+  // const is_bing = (dicturl.indexOf('www.bing.com') != -1);
```

Comment:
Shouldn't be commented out, but the existing dictionaries stored in the db should change.